### PR TITLE
chore(`@astrojs/underscore-redirects`): use Node.js for testing

### PR DESCRIPTION
**Pull request by at-the-vr**
## Changes

- Move [`@astrojs/underscore-redirects`](https://github.com/withastro/astro/tree/main/packages/underscore-redirects?rgh-link-date=2024-01-30T10%3A58%3A53Z) tests from chai to nodejs
- `chai` and `mocha` dependencies removed
- Related Issue #9873 

## Testing

`pnpm test --filter @astrojs/underscore-redirects` (All tests passing)

## Docs

Not needed

### CODE REVIEW

**Comment by at-the-vr** [comment]
Ubuntu Image breaking right after Apple declares macos-14. Coincidence, I think not 🤓

File: packages/underscore-redirects/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "build:ci": "astro-scripts build \"src/**/*.ts\"",
     "postbuild": "astro-scripts copy \"src/**/*.js\"",
     "dev": "astro-scripts dev \"src/**/*.ts\"",
-    "test": "mocha --exit --timeout 20000"
+    "test": "astro-scripts test \"test/**/*.test.js\""
   },
   "devDependencies": {
     "@types/chai": "^4.3.10",
```
Comment (bluwy) [inline packages/underscore-redirects/package.json:31]: Looks like we can remove the mocha and chai dev dependencies here, which seems to be the reason the CI is failing.

Comment (at-the-vr) [reply packages/underscore-redirects/package.json:31]: on it, thanks for the follow up 👍

**Review by ematipico — APPROVED** [review]

File: packages/underscore-redirects/package.json [hunk 2]
```diff
@@ -25,15 +25,11 @@
...
     "postbuild": "astro-scripts copy \"src/**/*.js\"",
     "dev": "astro-scripts dev \"src/**/*.ts\"",
-    "test": "mocha --exit --timeout 20000"
+    "test": "astro-scripts test \"test/**/*.test.js\""
   },
   "devDependencies": {
-    "@types/chai": "^4.3.10",
-    "@types/mocha": "^10.0.4",
     "astro": "workspace:*",
     "astro-scripts": "workspace:*",
```
Comment (ematipico) [inline packages/underscore-redirects/package.json:32]:
```suggestion
    "astro-scripts": "workspace:*"
```

**Comment by bluwy** [comment]
The lockfile also needs to be updated.

**Pull request merged**